File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/DemageEmblem.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;

public class DemageEmblem extends Equipable {
	{
		name = Locale.get("demage_emblem");
		description = Locale.get("demage_emblem_desc");
		sprite = "item-more_defense_and_damage";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.defenseModifier += 0.2f;
		this.owner.damageModifier += 0.2f;
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.defenseModifier -= 0.2f;
		this.owner.damageModifier -= 0.2f;
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/DemageEmblem.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class DemageEmblem : Equipable() {
	init {
		name = Locale.get("demage_emblem")
		description = Locale.get("demage_emblem_desc")
		sprite = "item-more_defense_and_damage"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.defenseModifier += getMod() / 100f
		this.owner.damageModifier += getMod() / 100f
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.defenseModifier -= getMod() / 100f
		this.owner.damageModifier -= getMod() / 100f
	}

	private fun getMod(): Float {
		return this.level * 20f
	}

	override fun getDescription(): String {
		return super.getDescription().replace("{PERCENT}", getMod().toInt().toString())
	}
}